src/services/expense.service.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const { Expense } = require('../../db');
+const { filterExpense } = require('../utils/filterExpense');
+
+const getAll = async(userId, categories, from, to) => {
+  const expenses = await Expense.findAll();
+
+  return filterExpense(expenses,
+    {
+      userId,
+      categories,
+      from,
+      to,
+    });
```

Comment:
Query from the database only expenses that you actually need instead of querying all and then filtering them.
https://sequelize.org/docs/v6/core-concepts/model-querying-basics/